Client/mainwindow.cpp

```diff
@@ -0,0 +1,234 @@
+// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
+#include <QMessageBox>
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+#include "SMTPClient.h"
+
+MainWindow::MainWindow(QWidget* parent)
+    : QMainWindow(parent)
+    , ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);   
+    
+    m_own_server_address    = m_xml.GetIpAddress();
+    m_own_server_port       = std::to_string(m_xml.GetListenerPort());
+    m_server_choice         = m_own_server_address;
+
+    connect(ui->ownServerButton, SIGNAL(clicked()), this, SLOT(OwnServerButtonClicked()));
+    connect(ui->gmailServerButton, SIGNAL(clicked()), this, SLOT(GmailServerButtonClicked()));
+    connect(ui->sendButton, SIGNAL(clicked()), this, SLOT(SendButtonClicked()));
+    connect(ui->exitButton, SIGNAL(clicked()), this, SLOT(ExitButtonClicked()));
+}
+
+MainWindow::~MainWindow()
+{        
+    delete ui;
+}
+
+void MainWindow::SendButtonClicked()
+{    
+    if (!m_own_server_clicked && !m_gmail_server_clicked)
```

Comment:
I would be good to set default server, and do not allow user not use empty smtp server